flink-state-backends/flink-statebackend-forst/src/main/java/org/apache/flink/state/forst/ForStMapState.java

```diff
@@ -139,6 +140,7 @@ public byte[] serializeValue(UV value) throws IOException {
     public UV deserializeValue(byte[] valueBytes) throws IOException {
         DataInputDeserializer inputView = valueDeserializerView.get();
         inputView.setBuffer(valueBytes);
+        inputView.readBoolean();
```

Comment:
We should respect this read value and return null if `true` (since it may come from the sync state). The invokers of this method should also deal with the returned null.